http/httpclient/clientBuilder.go

```diff
@@ -80,6 +86,12 @@ func (builder *httpClientBuilder) AddClientCertToTransport(transport *http.Trans
 func (builder *httpClientBuilder) Build() (*HttpClient, error) {
 	var err error
 	var transport *http.Transport
+
+	if builder.httpClient != nil {
+		// Using a custom http.Client, pass-though.
+		return &HttpClient{client: builder.httpClient, ctx: builder.ctx, retries: builder.retries}, nil
+	}
```

Comment:
Let's put this if statement at the very beginning of the method, because the two variables' definition are not needed for this if statement.
Let's change - 
```
	var err error
	var transport *http.Transport

	if builder.httpClient != nil {
		// Using a custom http.Client, pass-though.
		return &HttpClient{client: builder.httpClient, ctx: builder.ctx, retries: builder.retries}, nil
	}
``` 
to
```
	if builder.httpClient != nil {
		// Using a custom http.Client, pass-though.
		return &HttpClient{client: builder.httpClient, ctx: builder.ctx, retries: builder.retries}, nil
	}

	var err error
	var transport *http.Transport
```